pkg/server/apiextensions.go

```diff
@@ -298,41 +405,51 @@ func findCRD(crdName string, crds []*apiextensionsv1.CustomResourceDefinition) (
 // we can supply our own inheritance-aware CRD lister.
 type kcpAPIExtensionsSharedInformerFactory struct {
 	apiextensionsexternalversions.SharedInformerFactory
-	workspaceLister tenancylisters.ClusterWorkspaceLister
+	kcpClusterClient kcpclientset.ClusterInterface
```

Comment:
(No action required) Given that these fields are common across multiple types in this file, maybe define in a common type and embed?